apk: Implement apk list --full patch

Ansuel opened the pull request:
Implement apk list --full patch to mimik opkg package info.

---

@aparcar @hauke

Ansuel committed:
apk: Implement apk list --full patch

Implement apk list --full patch to mimik opkg package info.

Link: https://github.com/openwrt/openwrt/pull/16759
Link: https://github.com/openwrt/openwrt/pull/16759
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

aparcar commented:
This breaks both the buildbot (https://buildbot.aparcar.org/phase1/#/) as well as my local build:

```
In file included from ../src/app_list.c:15:
In file included from ../src/apk_applet.h:16:
../src/apk_database.h:198:26: warning: field 'overlay_tmpl' with variable sized type 'struct apk_package_tmpl' not at the end of a struct or class is a GNU extension [-Wgnu-variable-sized-type-not-at-end]
  198 |         struct apk_package_tmpl overlay_tmpl;
      |                                 ^
../src/app_list.c:128:34: error: no member named 'num' in 'struct apk_dependency_array'
  128 |         if (apk_array_len(pkg->depends->num)) {
      |                           ~~~~~~~~~~~~  ^
../src/apk_defines.h:229:48: note: expanded from macro 'apk_array_len'
  229 | #define apk_array_len(array)            _apk_array_len(&(array)->hdr)
      |                                                          ^~~~~
../src/app_list.c:134:66: error: no member named 'num' in 'struct apk_dependency_array'
  134 |                         printf("%s%s", d->name->name, i < apk_array_len(pkg->depends->num) ? ", ": "\n");
      |                                                                         ~~~~~~~~~~~~  ^
../src/apk_defines.h:229:48: note: expanded from macro 'apk_array_len'
  229 | #define apk_array_len(array)            _apk_array_len(&(array)->hdr)
      |                                                          ^~~~~
../src/app_list.c:137:35: error: no member named 'num' in 'struct apk_dependency_array'
  137 |         if (apk_array_len(pkg->provides->num)) {
      |                           ~~~~~~~~~~~~~  ^
../src/apk_defines.h:229:48: note: expanded from macro 'apk_array_len'
  229 | #define apk_array_len(array)            _apk_array_len(&(array)->hdr)
      |                                                          ^~~~~
../src/app_list.c:143:67: error: no member named 'num' in 'struct apk_dependency_array'
  143 |                         printf("%s%s", d->name->name, i < apk_array_len(pkg->provides->num) ? ", ": "\n");
      |                                                                         ~~~~~~~~~~~~~  ^
../src/apk_defines.h:229:48: note: expanded from macro 'apk_array_len'
  229 | #define apk_array_len(array)            _apk_array_len(&(array)->hdr)
      |                                                          ^~~~~
2 warnings and 4 errors generated.
ninja: build stopped: subcommand failed.
make[2]: *** [Makefile:87: /Users/user/src/openwrt/openwrt/build_dir/hostpkg/apk-3.0.0_pre20240909/.built] Error 1
make[2]: Leaving directory '/Users/user/src/openwrt/openwrt/package/system/apk'
time: package/system/apk/host-compile#5.95#1.93#9.97
    ERROR: package/system/apk [host] failed to build.
make[1]: *** [package/Makefile:175: package/system/apk/host/compile] Error 1
make[1]: Leaving directory '/Users/user/src/openwrt/openwrt'
make: *** [/Users/user/src/openwrt/openwrt/include/toplevel.mk:248: package/apk/host/compile] Error 2
```

@systemcrash how did you test this?

Ansuel committed:
apk: Bump to version 2024-11-08

Bump APK to version 2024-11-08, multiple fixup and pending arch patch
for ARM Big Endian merged upstream.

Link: https://github.com/openwrt/openwrt/pull/16759
Link: https://github.com/openwrt/openwrt/pull/16759
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
